src/contentScript/contentScript.tsx

```diff
@@ -2,17 +2,30 @@ import React, { useEffect, useState } from "react";
 import ReactDOM from "react-dom";
 
 import { Messages } from "../utils/messages";
+import { NoteType } from "../types/noteType";
+import { addNoteToStorage, fetchNotes } from "../utils/storage";
 import "./contentScript.css";
 
 import Note from "../components/Note";
 
 const App: React.FC<{}> = () => {
-  const [isActive, setIsActive] = useState<boolean>(false);
+  const [notes, setNotes] = useState<NoteType[]>([]);
+  const [currentPageUrl, setCurrentPageUrl] = useState<string>("");
 
-  const handleMessages = ({ type }) => {
-    console.log(type);
+  const handleMessages = async ({ type, data }) => {
     if (type === Messages.NEW_NOTE) {
-      setIsActive(!isActive);
+      const newNote = await addNoteToStorage(currentPageUrl);
+      setNotes((prevNotes) => [...prevNotes, newNote]);
+    }
+    if (type === Messages.NEW_PAGE) {
+      setCurrentPageUrl(data.url);
+      const notes = await fetchNotes(data.url);
+
+      setNotes(notes);
+    }
+    if (type === Messages.DELETE_ALL) {
+      chrome.storage.sync.remove(currentPageUrl);
+      setNotes([]);
```

Comment:
can we do it using switch?

Reply:
yep will do, and i thinking to make separate functions for each case but i dont know if we can pass useState func as params

Reply:
we can pass setSomethig function if we need